knebind/solver/solver.go

```diff
@@ -336,7 +448,7 @@ func (s *solver) buildLink2Links(d2n map[*opb.Device]*tpb.Node) map[*devLink][]*
 }
 
 func (s *solver) nodeMatches(dev *opb.Device, isATE bool) ([]*tpb.Node, error) {
-	var nodeList []*tpb.Node
+	nodeList := []*tpb.Node{}
```

Comment:
revert this line. Go style is let slices initialize to nil. Nearly all code and operations on slices don't distinguish a nil slice from an empty slice